include/envoy/network/listener.h

```diff
@@ -83,6 +86,12 @@ class ListenerConfig {
    */
   virtual bool bindToPort() PURE;
 
+  /**
+   * @return bool if 'true', the listener creates one socket with SO_REUSEPORT for each
+   * worker thread; otherwise, all worker threads share one socket.
+   */
+  virtual bool reusePort() PURE;
```

Comment:
> IMO if reusePort is always true for both TCP and UDP, this can be removed, otherwise, it's better to keep this interface. E.g. when calling ListenerComponentFactory::createListenSocket(), reuse_port is needed, if it's true, duplicate socket from hot restart parent process should be skipped.

Socket factory implementation already latches this value and ListenerComponentFactory::createListenSocket() is called from there. So I don't think we need this interface in ListenerConfig. But I think we should finalize the reuse port socket factory implementation first before thinking about what interface it needs to create a socket.

> Actually I think some other methods in Envoy::Server::ListenerImpl can be lift to ListenerConfig interface, e.g address(), socket type, name(). basically theses information are from envoy::api::v2::Listener.

We prefer not to make the interface more complicated unless it's necessary.

Reply:
reusePort() removed from ListenerConfig